va/win32/va_win32.c

```diff
@@ -0,0 +1,323 @@
+        free(pDisplayContext->pDriverContext->native_dpy);
+
+    if (pDisplayContext->pDriverContext)
+        free(pDisplayContext->pDriverContext);
+
+    VADisplayContextWin32* pWin32Ctx = (VADisplayContextWin32*) pDisplayContext->opaque;
+    if (pWin32Ctx)
+        free(pWin32Ctx);
+
+    free(pDisplayContext);
+}
+
+static VAStatus va_DisplayContextGetNumCandidates(
+    VADisplayContextP pDisplayContext,
+    int *num_candidates
+)
+{
+    /*  Always report the default driver name
+        If available, also add the adapter specific registered driver */
+    LUID* adapter = pDisplayContext->pDriverContext->native_dpy;
+    VADisplayContextWin32* pWin32Ctx = (VADisplayContextWin32*) pDisplayContext->opaque;
+    if(adapter && pWin32Ctx->registry_driver_available_flag)
+        *num_candidates = 2;
+    else
+        *num_candidates = 1;
+
+    return VA_STATUS_SUCCESS;
+}
+
+static VAStatus va_DisplayContextGetDriverNameByIndex(
+    VADisplayContextP pDisplayContext,
+    char **driver_name,
+    int candidate_index
+)
+{
+    LUID* adapter = pDisplayContext->pDriverContext->native_dpy;
+    VADisplayContextWin32* pWin32Ctx = (VADisplayContextWin32*) pDisplayContext->opaque;
+    *driver_name = calloc(sizeof(pWin32Ctx->default_driver_name), sizeof(char));
+    if(adapter && pWin32Ctx->registry_driver_available_flag)
+    {
+        /* Always prefer the adapter registered driver name as first option */
+        if(candidate_index == 0)
+            memcpy(*driver_name, pWin32Ctx->registry_driver_name, sizeof(pWin32Ctx->registry_driver_name));
+        /* Also provide the default driver name as a fallback option */
+        if(candidate_index == 1)
+            memcpy(*driver_name, pWin32Ctx->default_driver_name, sizeof(pWin32Ctx->default_driver_name));
+    }
+    else
+    {
+        /* Provide  the default driver name as a fallback option */
+        memcpy(*driver_name, pWin32Ctx->default_driver_name, sizeof(pWin32Ctx->default_driver_name));
+    }
+
+    return VA_STATUS_SUCCESS;
+}
+
+VADisplay vaGetDisplayWin32(
+    /* Can be null for adapter autoselection in the VA driver */
+    const LUID* adapter_luid
+)
+{
+    VADisplayContextP pDisplayContext;
+    VADriverContextP  pDriverContext;
+
+    pDisplayContext = va_newDisplayContext();
+    if (!pDisplayContext)
+        return NULL;
+
+    pDisplayContext->vaIsValid       = va_DisplayContextIsValid;
+    pDisplayContext->vaDestroy       = va_DisplayContextDestroy;
+    pDisplayContext->vaGetDriverNameByIndex = va_DisplayContextGetDriverNameByIndex;
+    pDisplayContext->vaGetNumCandidates = va_DisplayContextGetNumCandidates;
+    pDisplayContext->opaque = calloc(1, sizeof(VADisplayContextWin32));
+
+    VADisplayContextWin32* pWin32Ctx = (VADisplayContextWin32*) pDisplayContext->opaque;
+
+    /*  Initialize default_driver_name to the DX12 VA driver implementation
+        which will be provided when the adapter LUID doesn't have an associated VA driver name
```

Comment:
Please use consistent comment format - the license blurb one is seems the defacto standard.